exporters/zipkin/src/main/java/io/opentelemetry/exporters/zipkin/ZipkinSpanExporter.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.exporters.zipkin;
+
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.common.AttributeValue.Type;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.trace.Span.Kind;
+import io.opentelemetry.trace.Status;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import zipkin2.Endpoint;
+import zipkin2.Span;
+import zipkin2.codec.SpanBytesEncoder;
+import zipkin2.reporter.Sender;
+
+/**
+ * This class was based on the OpenCensus zipkin exporter code at
+ * https://github.com/census-instrumentation/opencensus-java/tree/c960b19889de5e4a7b25f90919d28b066590d4f0/exporters/trace/zipkin
+ */
+final class ZipkinSpanExporter implements SpanExporter {
+
+  private static final Logger logger = Logger.getLogger(ZipkinSpanExporter.class.getName());
+
+  // The naming follows Zipkin convention. As an example see:
+  // https://github.com/openzipkin/brave/blob/eee993f998ae57b08644cc357a6d478827428710/instrumentation/http/src/main/java/brave/http/HttpTags.java
+  // Note: these 3 fields are non-private for testing
+  static final String STATUS_CODE = "otel.status_code";
```

Comment:
I'm almost certain we wouldn't add 'otel' as a tag as that would just require re-mapping it back later.
As far as I can tell, this is an RPC code, and more specifically gRPC.

I don't think we would waste indexing on the one called code either. so we should pick one.

Let's drop "otel.status_code" and rename "otel.status_description" to "grpc.status_code",
as it seems to be the same value as what we tag today.

Ex. today grpc only tags "error" "grpc.status_code", ex "grpc.status_code" -> "UNIMPLEMENTED" (not a numerical tag)

This trades short term conflation with portability with existing search and indexing especially as I doubt
many RPC frameworks except grpc are integrated with otel at the moment.

You can add a better comment which is that we are tentatively re-using grpc until a final decision is made on rpc tags https://github.com/openzipkin/brave/pull/999

It may also be worth mentioning in the README that while the "error" tag is a zipkin tag, "grpc.status_code" is tentative. Adding the best tentative choice is better than blocking this work or introducing a non-existent tag that will add burden to existing sites as many don't have means to re-map tags (never needed to).

Reply:
This is definitely not just grpc. Here are the official mappings for http status codes:  https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md#status

@trask can comment as to whether those are being faithfully followed in the auto-instrumentation project or not.

With the naming of the tags...I just copied these from the opencensus version, and replaced "opencensus" with "otel". Happy to make them whatever is appropriate.

Reply:
if there's nothing besides grpc and http defined, then check for "http.status_code" first? if that's missing, assume it isn't and set to "otel.status_description" -> "grpc.status_code".

Reply:
re: auto-instrumentation - for instrumentation other than http and grpc we're only using Status.OK/Status.UNKNOWN (to mark success/failure). hmm, even for http and grpc instrumentation, we're only using Status.OK/Status.UNKNOWN, but that needs to be fixed (heading over there to file an issue now...)

Reply:
ok, I took a stab at this, @adriancole . Feedback welcome. Assuming grpc feels weird to me, but you're the boss!

Reply:
feeling weird is good. this is the tension that leads to better decisions later. please do let me know if you find anything besides grpc that uses this (besides the somewhat arbitrary lossy conversion of http status). meanwhile probably by month end we'll have an rpc tag and replace this.

Reply:
I guess my "weird feeling" might be assuaged if we removed the `g` and just said `rpc`, but maybe that's splitting hairs a bit.

Reply:
the issue I referenced will solve this probably before 0.4 or 0.5 here, just it is better to use a tag that exists today vs create one only to change it. don't worry.. I've the same opinion!

Reply:
in other words, we (OpenZipkin) should be able to settle the name, usage and content of the tag "grpc.status_code", just we owe ourselves to check this vs other RPC frameworks etc. I don't think this has been done here either, yet, so happy to share when that occurs.

Reply:
the comment above these values no longer makes sense. Do you have a better one to use until it gets settled?

Reply:
fixed per conversation below